### PR TITLE
chore(test-workflow): switch cache to v2

### PR DESCRIPTION
**Pull request by jbolda**
This enables just the cache change per #492.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -8,13 +8,18 @@ on:
...
+      - name: cache node_modules
+        uses: actions/cache@v2
+        # cache node_modules based on lockfile which can change
+        # when there is no explicit change in package.json
+        # and prevents cache bust when scripts or metadata is changed
+        # which does not effect the node_modules themselves
         with:
-          path: /home/runner/.cache/yarn/v6
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          path: '**/node_modules'
```
Comment (cowboyd) [inline .github/workflows/test.yml]: Shouldn't we have _both_ the `.cache` dir _and_ `node_modules`?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -8,13 +8,18 @@ on:
...
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
```
Comment (cowboyd) [inline .github/workflows/test.yml]: I like to have a key version as well which lets us just arbitrarily blow away the cache by swapping out keys in the event that we want to change our caching strategy without actually changing our `yarn.lock`, e.g.

```suggestion
          key: cachekey-v1-${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
```

File: .github/workflows/test.yml (hunk 1 above)
Comment (jbolda) [reply .github/workflows/test.yml]: does `.cache` provide anything extra from `node_modules`? I was hesitant to add more as there is a size limit to the overall cache and dropping too much in the cache will start to bust things unintentionally. As I say this though, there is a random timeout so maybe the cache is the source of this?

File: .github/workflows/test.yml (hunk 2 above)
Comment (jbolda) [reply .github/workflows/test.yml]: entirely reasonable ❤️

File: .github/workflows/test.yml (hunk 1 above)
Comment (cowboyd) [reply .github/workflows/test.yml]: @jbolda If I understand correctly, the `.cache` directory contains the downloaded tarballs, so in the event that you need to re-install a package, it doesn't have to download it. Actually, you're right. If we want to benefit fom this, we'll need to store it in a _separate_ cache, not tied to `yarn.lock`, basically one with a manually maintained cache key of `yarn-cache-v1` or something. That way, in the event that we do need to re-run a yarn install, we can skip the downloading step.

Maybe a separate PR though?

**Comment by cowboyd** [comment]
@jbolda Interesting. It looks like caching might be the cause of the intermittent failures? 😢

**Comment by cowboyd** [comment]
I wonder what happens if we even further limit our free variables by using the v1 cache action with `node_modules` and the `v2` cache action with `.cache`.

**Comment by jbolda** [comment]
I have been getting similar errors in windows which makes me think `node_modules` myself. I wonder if it's worth just trying v2 with `.cache`. I can add a later command that fails, and that cheeky fail will let us rerun it a handful of times.

**Comment by jbolda** [comment]
Looks like I can't rerun on my end, but that was successful so far.

**Comment by cowboyd** [comment]
> Looks like I can't rerun on my end, but that was successful so far.
I re-ran four times in a row without failure which seems pretty likely then that this is the problem.

**Comment by jbolda** [comment]
yeesh, maybe we skip that for now and merge everything else in over a few PRs and come back around to investigate?

**Comment by cowboyd** [comment]
Sounds like a plan!

**Comment by cowboyd** [comment]
Sounds like a plan!

**Review by cowboyd — APPROVED** [review]
Nice! I found out yesterday that our yarn.lock was out of sync. Wonder if that has anything to do wiith the failures we were seeing.

**Pull request merged**